src/firebase_functions/scheduler_fn.py

```diff
@@ -0,0 +1,131 @@
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Cloud functions to handle Schedule triggers."""
+
+import typing as _typing
+import dataclasses as _dataclasses
+import datetime as _dt
+import functools as _functools
+
+import firebase_functions.options as _options
+import firebase_functions.private.util as _util
+from functions_framework import logging as _logging
+from flask import (
+    Request as _Request,
+    Response as _Response,
+    make_response as _make_response,
+)
+
+# Export for user convenience.
+# pylint: disable=unused-import
+from firebase_functions.options import Timezone
+
+
+@_dataclasses.dataclass(frozen=True)
+class ScheduledEvent:
+    """
+    A ScheduleEvent that is passed to the function handler
+    """
+
+    job_name: str | None
+    """
+    The Cloud Scheduler job name.
+    Populated via the X-CloudScheduler-JobName header.
+    If invoked manually, this field is None.
+    """
+
+    schedule_time: _dt.datetime
+    """
+    For Cloud Scheduler jobs specified in the unix-cron format,
+    this is the job schedule time in RFC3339 UTC "Zulu" format.
+    Populated via the X-CloudScheduler-ScheduleTime header.
+
+    If the schedule is manually triggered, this field will be
+    the function execution time.
+    """
+
+
+_C = _typing.Callable[[ScheduledEvent], None]
+
+
+@_util.copy_func_kwargs(_options.ScheduleOptions)
+def on_schedule(**kwargs) -> _typing.Callable[[_C], _Response]:
+    """
+    Creates a handler for tasks sent to a Google Cloud Tasks queue.
+    Requires a function that takes a CallableRequest.
+
+    Example:
+
+    .. code-block:: python
+
+      from firebase_functions import scheduler_fn
+
+
+      @scheduler_fn.on_schedule(
+          schedule="* * * * *",
+          timezone=scheduler_fn.Timezone("America/Los_Angeles"),
+      )
+      def example(event: scheduler_fn.ScheduledEvent) -> None:
+          print(event.job_name)
+          print(event.schedule_time)
+
+
+    :param \\*\\*kwargs: ScheduleOptions options.
+    :type \\*\\*kwargs: as :exc:`firebase_functions.options.ScheduleOptions`
+    :rtype: :exc:`typing.Callable`
+            \\[ \\[ :exc:`firebase_functions.schedule_fn.ScheduledEvent` \\], :exc:`None` \\]
+            A function that takes a ScheduledEvent and returns nothing.
+    """
+    options = _options.ScheduleOptions(**kwargs)
+
+    def on_schedule_decorator(func: _C):
+
+        @_functools.wraps(func)
+        def on_schedule_wrapped(request: _Request) -> _Response:
+            schedule_time: _dt.datetime
+            schedule_time_str = request.headers.get(
+                "X-CloudScheduler-ScheduleTime")
+            if schedule_time_str is None:
+                schedule_time = _dt.datetime.utcnow()
+            else:
+                schedule_time = _dt.datetime.strptime(
+                    schedule_time_str,
+                    "%Y-%m-%dT%H:%M:%S%z",
```

Comment:
I think this is missing a trailing "Z".

IIUC, `datetime.fromisoformat()` could also work here?

Reply:
But then again - you tested the code in prod 🤔 wonder how it worked.

Reply:
Original code expected the Z in the test date time, in prod the dates were coming through without this and weren't an ISO format so this different parsing was needed; here's a log from prod from previous implementation showing the format: 

```
ValueError: time data '2023-04-13T10:19:00-07:00' does not match format '%Y-%m-%dT%H:%M:%SZ'

at ._strptime ( /layers/google.python.runtime/python/lib/python3.10/_strptime.py:349 )
at ._strptime_datetime ( /layers/google.python.runtime/python/lib/python3.10/_strptime.py:568 )
at .on_schedule_wrapped ( /layers/google.python.pip/pip/lib/python3.10/site-packages/firebase_functions/scheduler_fn.py:102 )
at .view_func ( /layers/google.python.pip/pip/lib/python3.10/site-packages/functions_framework/__init__.py:130 )
```

Reply:
That's good to know. Thanks for double checking!